sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/operation/LogDivertAppender.java

```diff
@@ -276,12 +276,19 @@ private static StringLayout initLayout(OperationLog.LoggingLevel loggingMode) {
     return getLayout(isVerbose, layout);
   }
 
-  public LogDivertAppender(OperationManager operationManager,
-    OperationLog.LoggingLevel loggingMode) {
+  public static LogDivertAppender create(OperationManager operationManager,
+                                         OperationLog.LoggingLevel loggingMode) {
+    CharArrayWriter writer = new CharArrayWriter();
+    return new LogDivertAppender(operationManager, loggingMode, writer);
+  }
+
+  private LogDivertAppender(OperationManager operationManager,
+    OperationLog.LoggingLevel loggingMode, CharArrayWriter writer) {
```

Comment:
Isn't it just to create `CharArrayWriter` in original constructor?

```suggestion
  public LogDivertAppender(OperationManager operationManager,
                                         OperationLog.LoggingLevel loggingMode) {
    CharArrayWriter writer = new CharArrayWriter();
```

Reply:
this is no way to create CharArrayWriter in original constructor.
because original constructor first need call supper constructor.  supper constructor need a CharArrayWriter as parameter.
```
private LogDivertAppender(OperationManager operationManager,
    OperationLog.LoggingLevel loggingMode, CharArrayWriter writer) {
    super("LogDivertAppender", initLayout(loggingMode), null, false, true, Property.EMPTY_ARRAY,
            new WriterManager(writer, "LogDivertAppender",
                    initLayout(loggingMode), true));
}
```

if create CharArrayWriter in original constructor like you suggest  compile will failed.

Reply:
Oh, yea. I didn't run the suggested one.